### PR TITLE
PHP DOM support and scripts for `php-8.2.0-slim`

### PR DESCRIPTION
**Pull request by gzurl**
This is a preliminary PR to discuss the method used for getting the _full-batteries_ and _slim_ versions.

I discarded the proposal of getting different folders (`slim`, `full`. etc.) inside the version number (e.g.: 8.2.0). It made really difficult to reuse current scripts.

Since we do not plan to have many flavors, I opted for duplicating per flavor, reusing as much as possible.

If we agree on this approach, we can replicate it in the other runtimes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,13 +18,13 @@ All you need in order to run these builds is to have `docker` or `podman` availa
 - `ruby/v3_2_0`
     - Resulting binaries are placed in `ruby/build-output/ruby`.
 
-### Build strategy
```
Comment (gzurl) [inline README.md:21]: In this file, only minor reading improvements.

File: libs/icu/release-72-1/wl-env-repo.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+if [[ $1 == "--unset" ]]
+then
+    unset WASMLABS_REPO
+    unset WASMLABS_REPO_NAME
+    return
+fi
+
+export WASMLABS_REPO=https://github.com/unicode-org/icu.git
+export WASMLABS_REPO_NAME=icu
+export WASMLABS_REPO_BRANCH=release-72-1
```
Comment (gzurl) [inline libs/icu/release-72-1/wl-env-repo.sh:12]: Needed to decouple branch names in the repo with whatever structure we want to have in WLR.

File: php/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ARG WASI_SDK_VERSION
 FROM ghcr.io/vmware-labs/wasmlabs/wasi-builder:${WASI_SDK_VERSION}
 RUN DEBIAN_FRONTEND=noninteractive apt install -y \
+      libtool-bin \
```
Comment (gzurl) [inline php/Dockerfile:4]: This is needed for building some libraries.

File: php/Dockerfile [hunk 4]
```diff
@@ -1,6 +1,7 @@
...
       bison \
       re2c \
       libsqlite3-dev \
```
Comment (gzurl) [inline php/Dockerfile:7]: I think this should be removed. But we would need to ensure CLFLAGS for PHP with SQLite points to or lib, and not to the headers installed into the system.

File: php/README.md [hunk 5]
```diff
@@ -6,6 +6,8 @@ You can build PHP by running the following Makefile targets:
 
 - `make php-7.3.33`
 - `make php-7.4.32`
+- `make php-8.1.11`
+- `make php-8.2.0`
```
Comment (gzurl) [inline php/README.md:10]: `make php-8.2.0-slim` is missing.

File: php/php-8.2.0-slim/patches [hunk 6]
```diff
@@ -0,0 +1 @@
+../php-8.2.0/patches
```
Comment (gzurl) [inline php/php-8.2.0-slim/patches:1]: It turned out that using a symbolic link was the best way to solve this issue of sharing patches between flavors.
- Link the entire patch folder if we reuse all the patches.
- If a flavor only needs a few, just link only them.
- Etc.

File: php/php-8.2.0-slim/wl-env-repo.sh [hunk 7]
```diff
@@ -9,4 +9,5 @@ fi
 
 export WASMLABS_REPO=https://github.com/php/php-src.git
 export WASMLABS_REPO_NAME=php
+export WASMLABS_REPO_BRANCH=php-8.2.0
```
Comment (gzurl) [inline php/php-8.2.0-slim/wl-env-repo.sh:13]: Now, these `wl-env-repo.sh` files are needed for each language-runtime-flavor.
So php-8.2.0 and php-8.2.0-slim have their own.

File: php/php-8.2.0-slim/wl-tag.sh [hunk 8]
```diff
@@ -0,0 +1 @@
+export WLR_TAG="php/8.2.0-slim"
```
Comment (assambar) [inline php/php-8.2.0-slim/wl-tag.sh:1]: Hm, shouldn't we have a single release tag for each version?
I mean, creating a php/8.2.0 tag should release both the slim and full version?

**Review by assambar — CHANGES_REQUESTED** [review]
Great work! 
Only two issues that trouble me:
 - Some vital lines seems commented out in the build of icu. Apparently it is OK, because the builds seem to be passing. So could you remove them?
 - Builds of Ruby and Python are failing.

File: libs/icu/release-72-1/wl-env-repo.sh (hunk 2 above)
Comment (gzurl) [reply libs/icu/release-72-1/wl-env-repo.sh:12]: As explained in our chat, this was needed to void a bigger refactor in all the other scripts :-)

File: libs/icu/release-72-1/wl-build.sh [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+if [[ -z "$WASMLABS_SKIP_CONFIGURE" ]]; then
+    export ICU_CONFIGURE='--enable-static --disable-shared'
+    logStatus "Configuring build with '${ICU_CONFIGURE}'... "
+    # ./runConfigureICU Linux --host=wasm32-wasi host_alias=wasm32-musl-wasi --target=wasm32-wasi target_alias=wasm32-musl-wasi ${ICU_CONFIGURE} || exit 1
+else
+    logStatus "Skipping configure..."
+fi
+
+# logStatus "Building... "
+# make || exit 1
```
[the remark this answers is not in the document]
Comment (gzurl) [reply libs/icu/release-72-1/wl-build.sh]: Added this:
```
# ICU headers are enough for some builds (e.g.: PHP).
# Uncomment for executing make and get the static libraries (lib/libicuXXXX.a)
`
``
So ICU v72.1 doesn't have the latest ./configure to recognize WASI. Apparently, PHP only requires the headers, we can leave it this way for now, and revisit it later (probably with a version > 72.1)

File: php/php-8.2.0-slim/wl-tag.sh (hunk 8 above)
Comment (gzurl) [reply php/php-8.2.0-slim/wl-tag.sh]: True. Fixed!

File: php/README.md (hunk 5 above)
Comment (gzurl) [reply php/README.md:10]: Fixed.

File: php/Dockerfile (hunk 4 above)
Comment (gzurl) [reply php/Dockerfile:7]: It seems PHP's `./configure` doesn't provide a `--with-sqlite3[=DIR]` knob or similar.
So, I guess we would need to keep using system libraries.

If really needed, we could set some symlinks in the container to point to our libraries, but I don't see the need for that right now.

File: libs/icu/release-72-1/wl-build.sh [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+if [[ ! -v WASMLABS_ENV ]]
+then
+    echo "Wasmlabs environment is not set"
+    exit 1
+fi
+
+# export CFLAGS_CONFIG="-O3 -g"
+export CFLAGS_CONFIG="-O0"
+
+export CFLAGS_WASI="--sysroot=${WASI_SYSROOT} -I./wasmlabs-stubs -D_WASI_EMULATED_MMAN -D_WASI_EMULATED_GETPID -D_WASI_EMULATED_SIGNAL -D_WASI_EMULATED_PROCESS_CLOCKS"
```
[the remark this answers is not in the document]
Comment (gzurl) [reply libs/icu/release-72-1/wl-build.sh]: Fixed

File: libs/libxml2/v2.10.3/wl-build.sh [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+if [[ ! -v WASMLABS_ENV ]]
+then
+    echo "Wasmlabs environment is not set"
+    exit 1
+fi
+
+# export CFLAGS_CONFIG="-O3 -g"
+export CFLAGS_CONFIG="-O0"
+
+export CFLAGS_WASI="--sysroot=${WASI_SYSROOT} -I./wasmlabs-stubs -D_WASI_EMULATED_MMAN -D_WASI_EMULATED_GETPID -D_WASI_EMULATED_SIGNAL -D_WASI_EMULATED_PROCESS_CLOCKS"
```
[the remark this answers is not in the document]
Comment (gzurl) [reply libs/libxml2/v2.10.3/wl-build.sh]: Fixed

**Review by assambar — APPROVED** [review]
Great!

File: scripts/wrappers/wasm-opt [hunk 12]
```diff
@@ -14,7 +14,6 @@ function logStatus {
 }
 
 if [[ -z "${WASMLABS_SKIP_WASM_OPT}" ]]; then
-   logStatus "wasm-opt (with args \"$@\") was intercepted; executing $BINARYEN_PATH/bin/wasm-opt"
```
Comment (ereslibre) [inline scripts/wrappers/wasm-opt:17]: Why remove this one? I think it's good to have this output

**Review by ereslibre — APPROVED** [review]
LGTM, thanks for addressing all the feedback. Just one optional additional comment to restablish the wrapper wasm-opt output when it's forwarding to the real wasm-opt.

File: scripts/wrappers/wasm-opt (hunk 12 above)
Comment (gzurl) [reply scripts/wrappers/wasm-opt:17]: Yes, the reason is I was getting too many of those messages and generating noise while I was debugging other stuff. I'm going to restore it and later on we can discuss how to improve all the tracing output.

File: libs/icu/release-72-1/wl-build.sh [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+if [[ ! -v WASMLABS_ENV ]]
+then
+    echo "Wasmlabs environment is not set"
+    exit 1
+fi
+
+# export CFLAGS_CONFIG="-O3 -g"
+export CFLAGS_CONFIG="-O0"
+
+export CFLAGS_WASI="--sysroot=${WASI_SYSROOT} -I./wasmlabs-stubs"
```
[the remark this answers is not in the document]
Comment (gzurl) [reply libs/icu/release-72-1/wl-build.sh]: Fixed!

**Pull request merged**